internal/clients/nightfall/nightfall.go

```diff
@@ -328,3 +334,35 @@ func (n *Client) ReviewDiff(
 		}
 	}
 }
+
+func filterFileDiffs(fileDiffs []*diffreviewer.FileDiff, fileIncludeList, fileExcludeList []string) []*diffreviewer.FileDiff {
+	if fileIncludeList != nil && len(fileIncludeList) > 0 {
+		fileDiffs = filterByFilePath(fileDiffs, fileIncludeList, true)
+	}
+	if fileExcludeList != nil && len(fileExcludeList) > 0 {
+		fileDiffs = filterByFilePath(fileDiffs, fileExcludeList, false)
+	}
+	return fileDiffs
+}
+
+func filterByFilePath(fileDiffs []*diffreviewer.FileDiff, regexPatterns []string, include bool) []*diffreviewer.FileDiff {
+	filteredFileDiffs := make([]*diffreviewer.FileDiff, 0, len(fileDiffs))
+	for _, fd := range fileDiffs {
+		matched := matchRegex(fd.PathNew, regexPatterns)
+		// if include (file inclusion), append if the filename matches a regex pattern
+		// if !include (file exclusion), append if the filename does not match any pattern
+		if (matched && include) || (!matched && !include) {
+			filteredFileDiffs = append(filteredFileDiffs, fd)
+		}
+	}
+	return filteredFileDiffs
+}
+
+func matchRegex(filePath string, regexPatterns []string) bool {
```

Comment:
did you have a chance to look into globbing vs regex? thoughts on doing one vs the other?

This library might be useful? https://golang.org/pkg/path/filepath/#Match

Reply:
Yup, if you look at the comment above, `filepath.Match` doesn't seem as user friendly for specifying subdirs. 

In addition, we probably can't use `filepath.Glob` (which uses the same expression syntax as `Match`) either, since it's looking locally for files that match the patterns, which doesn't translate to our use case where we're essentially filtering through a list of filenames.

Reply:
sorry which comment are you referring too?

And what are your thoughts on pros/cons of globbing vs. regex for this use case?

Reply:
oops it's on the `Conversation` tab

Reply:
But the main comparison is that globbing has better performance, and regex has easier usability (for users to specify in the config).